skimage/draw/draw3d.py

```diff
@@ -30,15 +40,56 @@ def ellipsoid(a, b, c, spacing=(1.0, 1.0, 1.0), levelset=False):
         Boolean dtype unless `levelset=True`, in which case a float array is
         returned with the level set above 0.0 representing the ellipsoid.
 
+    Notes
+    -----
+    The returned ellipsoid satisfies the following equation::
+        ((plane_rot / a) ** 2 + (row_rot / float(b)) ** 2 + (col_rot / float(c)) ** 2) = 1
+
+    where plane_rot, row_rot, and col_rot are the passive rotated coordinates generated using the Rotation matrix `R`::
+        [plane_rot, row_rot, col_rot].T = R[2,0,1].T @ [plane, row, col].T
+
+    where R is the standard active rotation matrix for the given Euler angles (theta, phi, psi) and coordinates (x, y, z) [1]_::
+        R = [[cos(psi)*cos(phi) - cos(theta)*sin(phi)*sin(psi), -sin(psi)*cos(phi) - cos(theta)*sin(phi)*cos(psi), sin(theta)*sin(phi)],
+             [cos(psi)*sin(phi) + cos(theta)*cos(phi)*sin(psi), -sin(psi)*sin(phi) + cos(theta)*cos(phi)*cos(psi), -sin(theta)*cos(phi)],
+             [sin(psi)*sin(theta), cos(psi)*sin(theta), cos(theta)]]
+
+    References
+    ----------
+    .. [1] Weisstein, Eric W. "Euler Angles." From MathWorld--A Wolfram Web Resource.
+            https://mathworld.wolfram.com/EulerAngles.html
+
     """
     if (a <= 0) or (b <= 0) or (c <= 0):
         raise ValueError('Parameters a, b, and c must all be > 0')
 
     offset = np.r_[1, 1, 1] * np.r_[spacing]
-
+    phi, theta, psi = rotation
+    # Define the standard rotation matrix for the given Euler angles defined for the coordinate system (x,y,z)
+    standard_rotation_matrix = np.array(
```

Comment:
There's some duplication with calls to `np.cos(psi)`, `np.cos(phi)`, ... going on here. Can we use sensibly named intermediate variables here? Might be good to streamline the whole creation into its own little helper function to make it reusable and easier to reason about. What do you think?

Reply:
moved it to an auxiallary function. How does that look?